clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -109,29 +109,45 @@
  * Identifiers for all the Kafka APIs
  */
 public enum ApiKeys {
-    PRODUCE(0, "Produce", ProduceRequest.schemaVersions(), ProduceResponse.schemaVersions()),
-    FETCH(1, "Fetch", FetchRequest.schemaVersions(), FetchResponse.schemaVersions()),
-    LIST_OFFSETS(2, "ListOffsets", ListOffsetRequest.schemaVersions(), ListOffsetResponse.schemaVersions()),
-    METADATA(3, "Metadata", MetadataRequest.schemaVersions(), MetadataResponse.schemaVersions()),
-    LEADER_AND_ISR(4, "LeaderAndIsr", true, LeaderAndIsrRequest.schemaVersions(), LeaderAndIsrResponse.schemaVersions()),
-    STOP_REPLICA(5, "StopReplica", true, StopReplicaRequest.schemaVersions(), StopReplicaResponse.schemaVersions()),
-    UPDATE_METADATA(6, "UpdateMetadata", true, UpdateMetadataRequest.schemaVersions(),
+    PRODUCE(0, "Produce", ProduceRequest.doc(),
```

Comment:
Did you consider writing the doc strings here instead of inside the request objects. Might be more convenient to find that way and it matches the same pattern that we use for `Errors` (for example).

Reply:
To me it seemed natural to keep that documentation with the request that way you can at least see what the request is composed of, OTOH doing it that way you still can't see the response. I'm happy to change if it you prefer?

Reply:
No strong preference, but I think it's fine if the high-level description of the API is decoupled from the request/response objects, which are mostly just concerned with schemas. The location in `ApiKeys` makes the descriptions a bit easier to find.

Reply:
@hachikuji I've now put the doc in `ApiKeys`.